frontend/js/src/artist-similarity/src/Data.tsx

```diff
@@ -0,0 +1,168 @@
+
+    return Color(red, green, blue);
+}
+
+const Data = () => {
+    const LIMIT_VALUE = 18;
+    const BASE_URL = "https://labs.api.listenbrainz.org/similar-artists/json?algorithm=session_based_days_7500_session_300_contribution_5_threshold_10_limit_100_filter_True_skip_30&artist_mbid=";
+    
+    var artist_mbid = "8f6bd1e4-fbe1-4f50-aa9b-94c450ec0f11";
+    var color1 = colorGenerator();
+    var color2 = colorGenerator();
+
+    const [similarArtists, setSimilarArtists] = useState({} as DatasetResponseType);
+    const [artist, setArtist] = useState({} as DatasetResponseType);
+    const [limit, setLimit] = useState(LIMIT_VALUE);
+    const [colors, setColors] = useState([color1, color2]);
+    
+    var transformedArtists = {};
+    var scoreList: Array<number> = [];
+    var similarArtistsList: Array<ArtistType> = [];
+    var mainArtist: ArtistType = {} as ArtistType;
+    var minScore = 0;
+    
+    const fetchData = (artist_mbid: string): void => {
+        fetch(BASE_URL + artist_mbid)
+        .then((response) => response.json())
+        .then((data) => processData(data))   
+    }
+    
+    const processData = (data: Array<ApiResponseType>): void => {
+        let mainArtist = data[1] as DatasetResponseType;
+        setArtist(mainArtist);
+
+        let similarArtists = data[3] as DatasetResponseType;
+        setSimilarArtists(similarArtists);
+        setColors([mixColor(colors[0], colors[1], 0.3), color2]);
+    }
+    
+    useEffect(() => {
+        fetchData(artist_mbid);
+        setColors([color1, color2]);
+    }, []);
+
+    similarArtistsList = similarArtists && similarArtists.data && (similarArtists.data.map((artist: any) => artist));
+    similarArtistsList = similarArtistsList && similarArtistsList.splice(0, limit);
+    
+    mainArtist = artist && artist.data && artist.data[0];
+    similarArtistsList && similarArtistsList.push(mainArtist);
+
+    // Calculating minScore for normalization which is always the last element of the array (because it's sorted)
+    minScore = similarArtistsList && similarArtistsList[LIMIT_VALUE - 1].score as number;
+    minScore = Math.sqrt(minScore);
+
+    transformedArtists = similarArtistsList && {
+        
+        "nodes": similarArtistsList.map((artist: ArtistType, index: number) => {
+
+            if(artist !== mainArtist) {
+                var computedScore = minScore / Math.sqrt(artist.score as number);
+                var computedColor = mixColor(colors[0], colors[1], (index / LIMIT_VALUE * computedScore));
+                scoreList.push(computedScore);
+            }
+            
+            else{
+                computedColor = colors[0];
+                similarArtistsList.pop();
+            }
+
+            return {
+                "id": artist.name,
+                "artist_mbid": artist.artist_mbid,
+                "size": artist.artist_mbid === mainArtist.artist_mbid ? 150 : 85,
```

Comment:
These constants should also be defined, not literals.

Reply:
I am not sure if I understand correctly, but I believe removing quotes should do the trick. Please correct me if I am wrong

Reply:
No, I was talking about the values 150 and 85. What are they?

Reply:
Ok. 150 and 85 are the node sizes. I have replaced them with appropriate constants MAIN_NODE_SIZE and SIMILAR_NODE_SIZE